discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -757,6 +756,73 @@ impl FeedbackScope for SqliteStorage {
         tx.commit().await?;
         Ok(document)
     }
+
+    async fn update_time_spent(
+        &self,
+        document: document::Id,
+        view_mode: ViewMode,
+        duration: Duration,
+    ) -> Result<TimeSpentDocumentView, Error> {
+        let view_time_ms = u32::try_from(duration.as_millis()).ok().unwrap_or(u32::MAX);
```

Comment:
🔧 this can be unwrapped directly, no need to convert it to an option first
```suggestion
        let view_time_ms = u32::try_from(duration.as_millis()).unwrap_or(u32::MAX);
```